### ISSUE TITLE
structuring to improve guessing

### ISSUE THREAD

**Issue by mcnesium** [jvenant/passff#127]
Here is a snippet of my current structure:

    myserver
    ├── ejabberd
    │   ├── mcnesium@myserver.mydomain.org
    …
    www
    ├── myserver.mydomain.org
    │   └── selfoss
    │       └── mcnesium
    …

Structure of the `mcnesium` file is as suggested in this Readme.md (pw, login, url, all lower case). I want it to guess the login for [selfoss](https://github.com/SSilence/selfoss/) when opening `myserver.mydomain.org/selfoss`, but it keeps selecting my jabber credentials. How can I improve this?

**Comment by oneirosFade** [comment]
I don't know if it is an option for you, but my setup for the same would be:
```
app
├── xmpp
...
www
├── selfoss
```
And the leaf nodes would contain "username: mcnesium@myserver.mydomain.org" and "username: mcnesium" respectively. It seems to work so far on my end.

**Comment by mcnesium** [comment]
Right, I do get a lot of false positives as well. Also, I do not only have `/selfoss` on the server, but for example `/owncloud`, `/roundcube` and a bunch of other tools, that need their own credentials. For now, it seems that the guessing only works for one login pair on one domain.

What about defining the structure like this:

    www (could be statically defined)
    ├── myserver.mydomain.org (domain must match)
    │   └── selfoss (a possible subdirectory of the domain)
    │       └── mcnesium (leaf is login, contains password in first line)

**Comment by jvenant** [comment]
I'm focused on the plugin new architecture for now. But I aware that the matching algorithm don't fit everybody needs.
I'll try to find something better after. I think I'm going to propose multiple algorithm based on the most used structures and also some kind of "custom algorithm" for advanced users where you will put your own javascript code.

**Comment by mcnesium** [comment]
Sounds great. I thought I was doing it wrong and therefore just looking for help, so don't feel being pushed. Thank you for sharing your work!
